Apps/TimelineApp/core/sensor/src/main/java/uk/ac/cam/cares/jps/sensor/source/network/SensorNetworkSource.java

```diff
@@ -92,13 +88,19 @@ public void sendPostRequest(String deviceId, byte[] compressedData, JSONArray se
                     // convert it to the correct format for the writeToDatabase method
                 Map<String, JSONArray> sensorDataMap = convertSensorDataToMap(sensorData);
 
-                    UnsentData unsentData = new UnsentData();
-                    String serializedData = serializeMap(sensorDataMap);
-                    unsentData.deviceId = deviceId;
-                    unsentData.data = serializedData;
-                    Log.e("network source", "data =" + serializedData);
-                    unsentData.timestamp = System.currentTimeMillis();
-                    sensorLocalSource.insertUnsentData(unsentData);
+                    String dataHash = sensorLocalSource.computeHash(sensorData.toString());
```

Comment:
if computeHash is used by both source, maybe can consider to put it in a util class as static function. It feels weird to put in either of the source class